File: dependencies/csharp/Dependencies/DependenciesTests/ListWrapperWithFactoryMethodTest.cs
using Dependencies;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DependenciesTests
{
    [TestClass]
    public class ListWrapperWithFactoryMethodTest
    {
        [TestMethod]
        public void TryTestingTheSizeOfAWrappedList()
        {
            var testedListWrapper = new ControlledListWrapper();
            Assert.AreEqual(5, testedListWrapper.WrappedListSize);
        }
    }

    class ControlledListWrapper : ListWrapperWithFactoryMethod 
    {
        protected override IList<int> CreateWrappedList()
        {
            return new List<int> { 1, 2, 3, 4, 5 };
        }
    }
}

File: dependencies/java/Dependencies/src/test/java/util/ListWrapperWithFactoryMethodTest.java
package util;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class ListWrapperWithFactoryMethodTest {
    @Test
    public void tryTestingTheSizeOfAWrappedList() {
        ListWrapperWithFactoryMethod testedListWrapper
                = new ListWrapperWithFactoryMethod() {
            protected List<Integer> createWrappedList() {
                return Arrays.asList(1, 2, 3, 4, 5);
            }
        };
        assertEquals(5, testedListWrapper.getWrappedListSize());
    }
}